### ISSUE TITLE
[BUG]: Bug in getting the output back.

### ISSUE THREAD

**Issue by landon2022** [blib-la/runpod-worker-comfy#65]
**Describe the bug**
I the source code: 
def process_output_images(outputs, job_id):
    output_images = {}

    for node_id, node_output in outputs.items():
        if "images" in node_output:
            for image in node_output["images"]:
                   output_images = os.path.join(image["subfolder"], image["filename"])

    print(f"runpod-worker-comfy - image generation is done")

    # expected image output folder
    local_image_path = f"{COMFY_OUTPUT_PATH}/{output_images}"

The outputs will also include the temp preview output images, and when loop through all the outputs somtimes you might get temp image as output, thus cause no image not found error at the end.
[
<img width="670" alt="Screenshot 2024-09-08 at 11 02 42" src="https://github.com/user-attachments/assets/af622382-f3e5-4e76-81c5-13d0f658721a">
<img width="670" alt="Screenshot 2024-09-08 at 11 07 47" src="https://github.com/user-attachments/assets/44b76917-e9a5-411a-a5c1-d9eb6d9327a7">
<img width="670" alt="Screenshot 2024-09-08 at 11 08 06" src="https://github.com/user-attachments/assets/27ee349f-02ec-4046-9676-ed9d0f4c8442">
](url)

**Comment by billyberkouwer** [comment]
I came up against this issue too - it caused some real headaches for my project. I found it sometimes returned the filename with a temp substring in it (e.g. COMFY_temp_esunj_0001_.png or  COMFY_temp_brqas_0008_.png). 

My hacky solution for this was to edit the Python code to include a function that checks for whether the filename includes a substring that follows this pattern and to remove it from the filename, because the COMFY_0008_.png image does exist. I'm a Javascript developer not normally working in Python so it was a challenge to debug the source of the issue, but even though it isn't not a great solution it has worked for me since. 

@landon2022 Your solution looks much cleaner and has found the source of the issue. Is the only addition you made to resolve this the line? -

`if image['type'] == "output":`

Or is there additional code needed to add the 'type' key/value to the output object?

**Comment by landon2022** [comment]
> I came up against this issue too - it caused some real headaches for my project. I found it sometimes returned the filename with a temp substring in it (e.g. COMFY_temp_esunj_0001_.png or COMFY_temp_brqas_0008_.png).
> 
> My hacky solution for this was to edit the Python code to include a function that checks for whether the filename includes a substring that follows this pattern and to remove it from the filename, because the COMFY_0008_.png image does exist. I'm a Javascript developer not normally working in Python so it was a challenge to debug the source of the issue, but even though it isn't not a great solution it has worked for me since.
> 
> @landon2022 Your solution looks much cleaner and has found the source of the issue. Is the only addition you made to resolve this the line? -
> 
> `if image['type'] == "output":`
> 
> Or is there additional code needed to add the 'type' key/value to the output object?

Yes. If there is only one Save image node in your workflow, this line of code would safice. But if your workflow contains more than one save image node, you need to change the source code to return all the saved images.

**Comment by TimPietruskyRunPod** [comment]
This issue is from 2024 and the entire handler / output pipeline has been rewritten since. The current response shape on 5.8.5 is documented at \`README.md\` (under "Example Response") and verified today as part of #157's triage:

\`\`\`json
{
  "id": "...",
  "status": "COMPLETED",
  "output": {
    "images": [{"filename": "ComfyUI_00001_.png", "type": "base64", "data": "..."}]
  }
}
\`\`\`

For S3 mode, each entry has \`type: "s3_url"\` and \`data\` is the URL.

Closing as stale. If you still hit a specific output retrieval bug on 5.8.5+, please open a fresh issue with: the request payload, the full response, and the worker logs.
